Dockerfile

```diff
@@ -1,16 +1,16 @@
-FROM rhel8/go-toolset:1.12.8 AS builder
+FROM rhel8/go-toolset:1.13.4 AS builder
 WORKDIR /opt/app-root/src/go/src/github.com/knative/client
 USER root
 COPY . .
-RUN TAG="v0.10.0" make build
```

Comment:
Why don't you want to maintain the tag from outside the Makfile in a single place ? We could go even so far to make it a Dockerfile ARG so that we can set it during the build (and using "latest" as a default).

Form where it the Dockerfile called ?

Reply:
Yes, the way TAG is fed in this setup could be improved.
The artifacts generated using this setup include:
 - p12n brew build (this Dockerfile at the root of repo is used)
 - kn image which gets synced at quay (once nightly for release-next and release-version image from release-branch)
            - this kn image is built at [`kn install`](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml#L11)
            - and copied over in the kn image [here](https://github.com/openshift/knative-client/blob/release-next/openshift/ci-operator/knative-images/client/Dockerfile#L4)
 - RPM: which uses `TAG=%{kn_cli_version} make build-cross`

so `make` command is being invoked from different places for building binary.

+1 to suggestion made above as, we can keep Makefile unaware of the tag but feed it from outside. 

For release-branches, I think the better place would be - right in the ci-operator config file [here](https://github.com/openshift/release/pull/6747/files#diff-ddfbd59ae93697247b6e6f81a3c1294fR11) updating the make command as

```
binary_build_commands: TAG=v0.11.0 make install
```
(this makes our release config yaml release-version specific while the release-next can simply invoke `make install`)

If this looks good - we can update the release-v0.11.0 ci-operator config and the RELEASE.md doc)

Reply:
1 - PR for updating release v0.11.0 ci-operator config to input TAG https://github.com/openshift/release/pull/6787

2 - Updated this PR (no changes in Makefile, Dockerfile is updated for brew)

3 - PR for recording these changes in RELEASE.md https://github.com/openshift/knative-client/pull/243